### PR TITLE
Add mobule pos_backend_partner

### PR DESCRIPTION
**Pull request by hparfr**
Choose a POS customer from the backend.

Needs #226

### CODE REVIEW

File: pos_backend_partner/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Akretion (http://www.akretion.com).
+# @author Raphaël Reverdy <raphael.reverdy@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Pos Backend customer",
+    "summary": "Choose point of sale customers in backend",
+    "version": "10.0.1.0.0",
+    "category": "Uncategorized",
```
Comment (legalsylvain) [inline pos_backend_partner/__manifest__.py]: category = Point of Sale

File: pos_backend_partner/README.rst [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+POS Backend Partner
+===================
+
+Choose a POS customer from the backend.
+
+
+- Replace customer selection screen in POS by a backend view.
+- Perf improvement : Full customer list is not downloaded anymore by the POS at startup.
+- Benefit from the backend features: advanced search, easy to extend views, etc.
+
```
Comment (legalsylvain) [inline pos_backend_partner/README.rst:16]: could you add a screenshot of the partner view in PoS ?

File: pos_backend_partner/README.rst [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+Configuration
+=============
+
+On each client's browser, during the first run, the permission for opening popup is prompted, it should be allowed.
+
+
+Roadmap
+=======
+
+
```
Comment (legalsylvain) [inline pos_backend_partner/README.rst]: Maybe add a know issue, about breaking off-line native mode ?

**Review by legalsylvain — COMMENTED** [review]
partial review for the time being, with cosmetic remarks. 

Note : If you add a temporary link to the other PR in the requirement, you'll make runbot green and users will have the possibility to live test your module.

thanks a lot. I'll take a more deeply look in some days.

File: pos_backend_partner/README.rst [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    :alt: License
+
+POS Backend Partner
+===================
+
+Choose a POS customer from the backend.
+
+
+- Replace customer selection screen in POS by a backend view.
+- Perf improvement : Full customer list is not downloaded anymore by the POS at startup.
```
Comment (yelizariev) [inline pos_backend_partner/README.rst:14]: But this actually means that POS will not work offline. I don't think that we can use word "improvement" here. It's a "change" or something like that

Comment (yelizariev) [reply pos_backend_partner/README.rst:14]: Well, it's a performance improvement, but at the expense of offline mode

Comment (yelizariev) [reply pos_backend_partner/README.rst:14]: Just found, that a note was added to base module:
https://github.com/OCA/pos/pull/226/commits/946882de2e503c3950c148a8309d386b1cb552a4

Still think that it has to be added here too

Comment (hparfr) [reply pos_backend_partner/README.rst:14]: It's a huge perf improvement if you have a lot of customers. In my case it's about 200 000 partners, and users have browsers warnings about slow and unresponsive scripts.

Offline mode still works, you just loose the ability to have a named customer while offline.

Readme has been corrected with your remarks.

Comment (yelizariev) [reply pos_backend_partner/README.rst:14]: Analog of pos_cache with customers caching may resolve this problem?

Comment (hparfr) [reply pos_backend_partner/README.rst:14]: When you introduce cache, you add a lot of complexity. 
Here it's much simplier : no list, nothing to wait, nothing to invalidate, always up to date.

Perfs is one nice part of the features, having advanced search and view customisations in one place reduce dev time.

**Review by sylvainc — APPROVED** [review]

**Review by bealdav — APPROVED** [review]
Just tested.
Works like a charm

Nice integration

File: pos_backend_partner/README.rst [hunk 5]
```diff
@@ -0,0 +1,75 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
```
Comment (elicoidal) [inline pos_backend_partner/README.rst]: Please update to latest template

File: pos_backend_partner/README.rst [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    :alt: License
+
+POS Backend Partner
+===================
+
+Choose a POS customer from the backend.
+
+
+- Replace customer selection screen in POS by a backend view.
+- Perf improvement : Full customer list is not downloaded anymore by the POS at startup.
```
Comment (elicoidal) [inline pos_backend_partner/README.rst:14]: This should be placed in the known limitations

File: pos_backend_partner/README.rst [hunk 7]
```diff
@@ -0,0 +1,75 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License
+
+POS Backend Partner
```
Comment (elicoidal) [inline pos_backend_partner/README.rst]: add a line before as document title

File: pos_backend_partner/README.rst [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+Configuration
+=============
+
+On each client's browser, during the first run, the permission for opening popup is prompted, it should be allowed.
+
+
+Roadmap
+=======
+
+- See pos_backend_communication
```
Comment (elicoidal) [inline pos_backend_partner/README.rst]: Add a link to the PR or module

File: pos_backend_partner/README.rst [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/pos/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/OCA/pos/issues/new?body=module:%20pos_backend_partner%0Aversion:%208.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (elicoidal) [inline pos_backend_partner/README.rst]: update to latest template

File: pos_backend_partner/README.rst [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+
+Credits
+=======
+
+Contributors
+------------
+
+* Raphaël Reverdy <raphael.reverdy@akretion.com>
+
+Maintainer
```
Comment (elicoidal) [inline pos_backend_partner/README.rst]: update to latest template

File: pos_backend_partner/__init__.py [hunk 11]
```diff
@@ -0,0 +1 @@
+from . import models
```
Comment (elicoidal) [inline pos_backend_partner/__init__.py]: utf-8 encoding (in all `__init__.py`)
add a line

File: pos_backend_partner/__manifest__.py [hunk 12]
```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Akretion (http://www.akretion.com).
+# @author Raphaël Reverdy <raphael.reverdy@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Pos Backend customer",
```
Comment (elicoidal) [inline pos_backend_partner/__manifest__.py]: POS

File: pos_backend_partner/__manifest__.py [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+    'installable': True,
+    "depends": [
+        "pos_backend_communication",
+    ],
+    "data": [
+        'views/assets.xml',
+        'views/backend_partner.xml',
+    ],
+    "qweb": [
+    ]
```
Comment (elicoidal) [inline pos_backend_partner/__manifest__.py]: remove empty keys

File: pos_backend_partner/models/partner.py [hunk 14]
```diff
@@ -0,0 +1,24 @@
+# coding: utf-8
+# @author Raphael Reverdy <raphael.reverdy@akretion.com>
```
Comment (elicoidal) [inline pos_backend_partner/models/partner.py]: add copyright entity

File: pos_backend_partner/static/src/js/back_partner.js [hunk 15]
```diff
@@ -0,0 +1,17 @@
+'use strict';
```
Comment (elicoidal) [inline pos_backend_partner/static/src/js/back_partner.js:1]: You can add copyright notice to js files too

**Review by elicoidal — CHANGES_REQUESTED** [review]
Formal details.
thanks for the contribution

**Comment by elicoidal** [comment]
Tests OK but I would remove the annoying pop-up "Customer Set"
To be noted that there is a glitch (non blocking) when you select multiple customers one after the others: in backend the button looks like "it is pushed" for all selected customers (should be only the last one).
screenshot from 2017-12-08 09-25-45

File: pos_backend_partner/__init__.py (hunk 11 above)
Comment (bealdav) [reply pos_backend_partner/__init__.py]: @elicoidal thanks for your review,
Here it's not utf-8, it's only ascii, python 2 support ascii by default without coding mentions
For v11 (python 3), all coding mentions 'll be rmoved as default utf-8
Yes it's in the guideline. But there is no added value for the future
Thanks for your comprehension

Comment (elicoidal) [reply pos_backend_partner/__init__.py]: Hi @bealdav I just follow the guidelines (https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#idioms).
When you migrate to v11, you will have to remove the coding line.

We need to stick to the guidelines for consistency over all modules.
I hope you understand

File: pos_backend_partner/README.rst (hunk 2 above)
Comment (hparfr) [reply pos_backend_partner/README.rst:16]: screenshot added

File: pos_backend_partner/README.rst (hunk 6 above)
Comment (hparfr) [reply pos_backend_partner/README.rst:14]: It's not a limitation:  it's the main feature ! :)

File: pos_backend_partner/__init__.py (hunk 11 above)
Comment (hparfr) [reply pos_backend_partner/__init__.py]: done

**Comment by hparfr** [comment]
about annoying pop-up "Customer Set" : it's needed to get the focus on the window. It's on the roadmap of the base module.

The css issue has been added to the roadmap of the present module.

Thanks for your review @elicoidal

File: pos_backend_partner/README.rst (hunk 6 above)
Comment (elicoidal) [reply pos_backend_partner/README.rst:14]: hehe OK :)

**Comment by legalsylvain** [comment]
@elicoidal, all your remarks has been taken into account. Is it ok for merging ?

**Review by elicoidal — APPROVED** [review]

**Comment by legalsylvain** [comment]
Hi. Take care when you merge the PR that temporary dependencies has been removed. Travis is broken since a month due to the merge of this PR (and the deletion of the akretion branch after).
I fixed it in #240.
regards.

**Pull request merged**